src/lightning_app/components/database/server.py

```diff
@@ -0,0 +1,196 @@
+import os
+from functools import partial
+from typing import List, Optional, Type, Union
+
+from fastapi import FastAPI
+from uvicorn import run
+
+from lightning import BuildConfig, LightningFlow, LightningWork
+from lightning_app.components.database.client import _DatabaseClientFlow, _DatabaseClientWork
+from lightning_app.components.database.utilities import (
+    create_database,
+    delete_database,
+    general_delete,
+    general_insert,
+    general_select_all,
+    general_update,
+    reset_database,
+)
+from lightning_app.utilities.app_helpers import Logger
+from lightning_app.utilities.imports import _is_sqlmodel_available
+
+if _is_sqlmodel_available():
+    from sqlmodel import SQLModel
+
+
+logger = Logger(__name__)
+engine = None
+
+
+class DatabaseType:
+    FLOW = "flow"
+    WORK = "work"
+
+
+class Database(LightningFlow):
```

Comment:
Should this component be part of lightning or as a separate component to be used in the lightning gallery? I would prefer to have it separately similar to the Redis component or a postgress in the future.

Reply:
I would prefer to keep it in the framework right now. I don't think the 1 package 1 component is really scalable.

Reply:
Furthermore, the Echo App copied the code over. I want to encourage folks to use core components and build upon them.

Reply:
Is it something that has been discussed and agreed to put similar components inside the framework ?

The component doesn't work out of the box, it needs to install its dependencies.

Reply:
Yes, we chatted with @lantiga in Bergamo about it.
Overall, core abstractions should live within the framework as it pushes for re-utilisation.

Good examples are ServeGradio and TracerPythonScript. They are being re-used in lot of codebases.

Reply:
awesome, thanks !